### DISCUSSION TITLE
Turn service on/off

### DISCUSSION THREAD

**Discussion by Mar-85** [kmicki/SteamDeckGyroDSU#38] [Ideas]
Is there an easy way to turn this service on and off when not needed? Can I create some shortcuts to terminal commands to do this. I realized I don't really need this running that often and would rather not have it constantly in the background

**Comment by kmicki** [comment]
To stop and disable auto start:
```
    systemctl --user stop sdgyrodsu
    systemctl --user disable sdgyrodsu
```

To start and enable auto start:
```
    systemctl --user enable --now sdgyrodsu
```

To start without enabling auto start:
```
    systemctl --user start sdgyrodsu
```

A remark: when there is no program (emulator) connected to sdgyrodsu, the only thing it does is wait for an UDP telegram on an open socket. It barely uses any resources. Of course if you want to use Cemu or something that connects to sdgyrodsu but you don't need the motion controls input, then stopping the service will save some CPU time.

**Reply by Mar-85** [reply]
Thanks very much for the comprehensive reply!

By auto start do you mean when the deck reboots the service will or will not start with a system start?

Ah, I was under the impression it's constantly sending out gyro movements. So if nothing is connected to UDP the service isn't sending out anything? This is good to know. Thanks very much for the hard work.

**Reply by kmicki** [reply]
Yes, that's what I mean by auto start.

DSU UDP clients are cyclically sending a packet to sdgyrodsu. The service take advantage of that and "goes to sleep" when no packet from any client is detected for some time (up to 10 seconds I think). You can test it by closing the emulator while watching the process monitor - sdgyrodsu's CPU usage will drop to zero after a few seconds.
